components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -474,6 +475,19 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
                         },
                     );
                     this.disposables.push(resetListener);
+
+                    const resetListenerFromRedis = this.subscriber.listenForWorkspaceInstanceUpdates(
+                        ws.ownerId,
+                        (ctx, instance) => {
+                            if (instance.id === wsi.id) {
+                                this.forwardInstanceUpdateToClient(ctx, instance);
```

Comment:
We would now send two messages to the listener, no? I saw that the subscriber doesn't send at all atm, but it still looks strange loking at this code. maybe already introduce the featureflag and ue it to forward either from redis or from message-bus?

Reply:
With the feature flag disabled, we'd register the listener, but not deliver the updates (yet). Currently we've got this enabled only on non-prod so this wouldn't be a problem.

We'll need to adjust the feature flag a bit to make it per message specific such that we can enable just workspace instance updates, but keep the other updates going from the current messagebus.

I can do a pre-PR for this change, if you feel the double registration of clients poses a problem, alternatively, we can land this and follow-up with the feature flag adjustment.

Reply:
It's ok to fix it afterward but wanted to highlight that the code looks like it does double notification without the external context.

Reply:
Yeah makes sense. It's the cost of shifting the flag into the `redis-subscriber` rather than around the invocation itself